### PR TITLE
Adding billing mode. (Used #574 as reference)

### PR DESCRIPTION
**Pull request by leobarcellos**
- This fixes #574 large diffs and add tests

### CODE REVIEW

File: pynamodb/tests/test_model.py [hunk 1]
```diff
@@ -580,9 +613,12 @@ def fake_wait(*obj, **kwargs):
                 'ProvisionedThroughput': {
                     'ReadCapacityUnits': 25, 'WriteCapacityUnits': 25
                 },
-                'TableName': 'UserModel'
+                'TableName': 'UserModel',
+                'BillingMode': 'PROVISIONED'
             }
             actual = req.call_args_list[1][0][1]
+            print(sorted(actual.keys()))
+            print(sorted(params.keys()))
```
Comment (ikonst) [inline pynamodb/tests/test_model.py]:
```suggestion
```

File: pynamodb/tests/test_model.py [hunk 2]
```diff
@@ -580,9 +613,12 @@ def fake_wait(*obj, **kwargs):
                 'ProvisionedThroughput': {
                     'ReadCapacityUnits': 25, 'WriteCapacityUnits': 25
                 },
-                'TableName': 'UserModel'
+                'TableName': 'UserModel',
+                'BillingMode': 'PROVISIONED'
             }
             actual = req.call_args_list[1][0][1]
+            print(sorted(actual.keys()))
```
Comment (ikonst) [inline pynamodb/tests/test_model.py]:
```suggestion
```

**Comment by edholland** [comment]
@ikonst: Is there anything outstanding on this PR? Lack of support for per-request billing mode is holding back some of our usecases for Pynamo

**Comment by ikonst** [comment]
@edholland Thanks for the ping, there's nothing outstanding and I'll merge it.

FWIW, since the billing mode merely affects table provisioning, this doesn't stop you from using PynamoDB to work with such tables, but merely stops you from provisioning tables with PynamoDB. As of now, PynamoDB falls short other tools to perform DynamoDB provisioning, e.g. Terraform or even doing it manually through the AWS console / aws-cli. For example, PynamoDB cannot add indexes to existing tables.

**Pull request merged**